### PR TITLE
Allow filtering of subjects selection boxes

### PR DESCRIPTION
**Pull request by abought**
# Ticket
https://openscience.atlassian.net/browse/PREP-126

# Purpose
Fix "filter search results" subject selector box.

screen shot 2016-08-22 at 2 56 53 pm

# Summary of changes
- Fix "subject picker" search filtering on search page
- Turn back on jshint curly brace checking and address linter issues

# Future
Will clean up and deduplicate these selector boxes later. For right now, just pushing a working version so we can move to higher priority tasks.

### CODE REVIEW

**Comment by chrisseto** [comment]
https://github.com/airbnb/javascript/blob/master/README.md#blocks--braces

**Comment by abought** [comment]
> https://github.com/airbnb/javascript/blob/master/README.md#blocks--braces

There are limits to even my love of style guides, but in this case the rule is functional rather than cosmetic. I find the JSHint argument far more compelling- a safeguard against potentially hard to trace bugs.

http://jshint.com/docs/options/#curly

>However, in some circumstances, it can lead to bugs (you'd think that sleep() is a part of the loop while in reality it is not):

File: app/components/file-uploader.js [hunk 1]
```diff
@@ -96,7 +96,9 @@ export default Ember.Component.extend({
         },
         complete(_, dropzone, file) {
             // Complete is called when swapping out files for some reason...
-            if (file.xhr === undefined) return;
+            if (file.xhr === undefined) {
```
Comment (icereval) [inline app/components/file-uploader.js]: @abought, please do not reformat code unless it does not meet the style guide. A single line if blocks is acceptable per the AirBnB style guide we are following.

https://github.com/airbnb/javascript/blob/master/README.md#blocks--braces

Comment (abought) [inline app/components/file-uploader.js]: > @abought, please do not reformat code unless it does not meet the style guide. A single line if blocks is acceptable per the AirBnB style guide we are following.

See reply to Chris' similar comment above.
https://github.com/CenterForOpenScience/ember-preprints/pull/51#issuecomment-241613945

**Comment by chrisseto** [comment]
https://github.com/CenterForOpenScience/ember-preprints/blob/develop/package.json#L76

**Comment by abought** [comment]
> https://github.com/CenterForOpenScience/ember-preprints/blob/develop/package.json#L76

If you can make jshint enforce the rules for multiple lines (but allow single), that's wonderful.

Unfortunately, turning the rule on for multiple lines (potential bug source) also required reformatting single lines. It seemed the lesser of two headaches.

**Comment by icereval** [comment]
JSHint makes many recommendations we will not be following, please stick w/ the AirBnB guide.

**Comment by abought** [comment]
Since there were no comments about the code or functionality, I've gone ahead and pushed the changes requested, and am going to bed.

**Pull request merged**